clubs/backends.py

```diff
@@ -0,0 +1,78 @@
+import uuid
+
+from accounts.settings import accounts_settings
+from django.contrib.auth import get_user_model
+from django.contrib.auth.backends import RemoteUserBackend
+
+
+class LabsUserBackend(RemoteUserBackend):
+    def authenticate(self, request, remote_user):
+        """
+        Authenticate a user given a dictionary of user information from
+        platform.
+        TEMPORARY CUSTOM BACKEND TO HANDLE MIGRATION TO PENNIDS
+        """
+        if not remote_user:
+            return
+        User = get_user_model()
+        pennkey_user_query = User.objects.filter(username=remote_user['username'])
+        if len(pennkey_user_query) == 0:
+            # User hasn't logged into Clubs before
+            user, created = User.objects.get_or_create(id=remote_user['pennid'])
+        else:
+            pennkey_user = pennkey_user_query[0]
+            # Create new user with correct id
+            if pennkey_user.id != remote_user['pennid']:
+                pennkey_user.username = uuid.uuid4()
+                pennkey_user.save()
+                user, created = User.objects.create(id=remote_user['pennid'], username=remote_user['username']), True
+
+                # Update all many-to-many relationships
+                for favorite in pennkey_user.favorite_set.all():
```

Comment:
Will need to inform all of the other developers about this, I actually recently added a `Profile` model which this new backend would break.

Ideally use Python introspection to identify all the many to many, one to one, and foreign key relationships, and switch them over this way if possible.

Reply:
So I didn't try to copy information from the `Profile` model because it didn't look like anything was using it yet, but if that's not the case, I can add that in. Currently, everything but profile and events is copied over.